### ISSUE TITLE
VS Code not responding on launching

### ISSUE THREAD

**Issue by RealShivesh** [microsoft/vscode#188397]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.80.1
- OS Version: Ubuntu 22.04

Steps to Reproduce:
1. Opening vscode from terminal or from desktop icon

I installed vscode from the snapstore, and while launching vscode, I am always encountering the issue: Visual Studio Code is not responding and this is happening everytime. 
I tried uninstalling, and reinstalling from the official website, ubuntu software store and even installed insiders, but still the same issue was persisting. 

Then after doing some digging, I tried running the command `code --verbose`, and this was the terminal response, with vscode launching with the same issue.

The verbose response didn't fit the character length in this issue comment.

**Comment by akingdom** [comment]
@RealShivesh, suggestion: try...

    code --verbose > code-verbose1.txt

...then upload that as a GIST or PROJECT and share the link

**Comment by hpst3r** [comment]
VS Code Version: 1.80.1
OS Version: Debian Sid (Trixie) fully updated (6.3.0-2-amd64)
Extensions: None (fresh installation of Code)

VSCode consistently hangs on launch after a fresh install (of the amd64 .deb) on a newly installed system. UI elements seem to fully load, then application becomes unresponsive. Does not seem to become responsive after ~ an hour. Unfortunately didn't grab verbose output from that run.

verbose output(s) for brief launches: https://gist.github.com/hpst3r/fc0598e67eeea0d28d3811a7d806a1ab

screengrab: https://youtu.be/P_mDkrYRMGY

inxi: https://gist.github.com/hpst3r/b1bdf7e66de7f969ad00dce423590721

**Comment by heyitsarpit** [comment]
@akingdom I had the same issue today, all windows and buttons are unresponsive and I can't close vscode either.

Here's my debug output - https://gist.github.com/heyitsarpit/aafb5d8d258ea2ee5eebcfb974f28115

**Comment by thumperward** [comment]
Seeing this as well, MacBookPro12,1 running a completely fresh install of Debian 12.1. 1.79.1-1686587647 is fine, everything after that hangs either immediately on window creation or a split-second afterwards (occasionally the left-hand icons show up and can be moused over, but usually most widgets don't get created before it hangs).

gist here: https://gist.github.com/thumperward/66f493d310c6c9c954e030def36cee41

**Comment by stodge** [comment]
I just installed vscode on a fresh install of Debian 12 and it always hangs on startup. The last thing logged is (if relevant):

`[8817:0909/210354.714141:INFO:CONSOLE(630)] "%cTRACE color: #888 [mainThreadSecretState] Getting password for vscode.github-authentication extension:  github.auth", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (630)`

**Comment by deepak1556** [comment]
This would be another case of https://github.com/microsoft/vscode/issues/190360, can affected users on linux check if the issue persists with latest stable `v1.82`

@stodge if you are already on `v1.82`, can you check if the issue is present when you start like `code --password-store="basic"`

**Comment by trentn** [comment]
On Ubuntu 22.04 with `v1.82`, I experienced the same issue as @stodge with the same log entry regarding GitHub authentication.
When started with `code --password-store="basic"`, as suggested, VSCode successfully starts up.

**Comment by stodge** [comment]
Sorry @deepak1556, I haven't had time to look at this yet.

**Comment by stodge** [comment]
> This would be another case of #190360, can affected users on linux check if the issue persists with latest stable `v1.82`
> 
> @stodge if you are already on `v1.82`, can you check if the issue is present when you start like `code --password-store="basic"`

I tried this and it starts without hanging.

**Comment by IGS-GIT** [comment]
bumping this. suffering the same issue on Zorin 16.3. everything is updated and a fresh version of code was installed. it just hangs and says not responding until it eventually loads. didn't do this before 1.82

**Comment by itsmenewbie03** [comment]
> On Ubuntu 22.04 with `v1.82`, I experienced the same issue as @stodge with the same log entry regarding GitHub authentication. When started with `code --password-store="basic"`, as suggested, VSCode successfully starts up.

is there any permanent fix for this?

**Comment by tonnyiowa10** [comment]
I found a file in ~/.argv.json where you can place your parms. to launch VSCode.
This is what I did 
"password-store" : "basic" 
and it seems to work for me.

**Comment by tonnyiowa10** [comment]
~/.vscode/argv.json
by the way, opps.

**Comment by stodge** [comment]
I had to modify my XFCE launcher to start vscode with —password-store=basic.

**Comment by itsmenewbie03** [comment]
> I found a file in ~/.argv.json where you can place your parms. to launch VSCode. This is what I did "password-store" : "basic" and it seems to work for me.

worked for me thanks

**Comment by thumperward** [comment]
1.82.2 no longer demonstrates this for me, without having to make any config changes.

**Comment by liviuflo** [comment]
1.82.3 hang-crashes on launch, on a fresh installation of Ubuntu 20.04, even with the `password-store` setting in place.
Manually installed the 1.81.1 .deb package and it seems to work just fine.

**Comment by Rinfore** [comment]
Faced the same error with v1.85.0. Using password-store allows a launch without getting stuck on "not responding"

**Comment by zyrridian** [comment]
i had the same issue after installing git on ubuntu, because when i try to uninstall git, everything's fine

**Comment by YangRongAtGit** [comment]
Same issue here, having a warning message `The window is not responding` when starting the VSCode on Ubuntu:
**VSCode info:**
1.85.2
8b3775030ed1a69b13e4f4c628c612102e30a681
x64

**OS info:**
LSB Version:	core-11.1.0ubuntu4-noarch:security-11.1.0ubuntu4-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy

```
[116594:0124/174223.646509:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.
[90m[main 2024-01-24T04:42:23.706Z][0m [File Watcher (node.js)] Request to start watching: /home/rongyang/.config/Code/User (excludes: <none>, includes: <all>, correlationId: <none>),/home/rongyang/.config/Code/User/settings.json (excludes: <none>, includes: <all>, correlationId: <none>)
[90m[main 2024-01-24T04:42:23.717Z][0m Starting VS Code
[90m[main 2024-01-24T04:42:23.717Z][0m from: /usr/share/code/resources/app
[90m[main 2024-01-24T04:42:23.717Z][0m args: {
  _: [],
  diff: false,
  merge: false,
  add: false,
  goto: false,
  'new-window': false,
  'reuse-window': false,
  wait: false,
  help: false,
  'list-extensions': false,
  'show-versions': false,
  'pre-release': false,
  version: false,
  verbose: true,
  status: false,
  'prof-startup': false,
  'no-cached-data': false,
  'prof-v8-extensions': false,
  'disable-extensions': false,
  'disable-gpu': false,
  'disable-chromium-sandbox': false,
  'ms-enable-electron-run-as-node': false,
  telemetry: false,
  debugRenderer: false,
  'enable-smoke-test-driver': false,
  logExtensionHostCommunication: false,
  'skip-release-notes': false,
  'skip-welcome': false,
  'disable-telemetry': false,
  'disable-updates': false,
  'use-inmemory-secretstorage': false,
  'disable-workspace-trust': false,
  'disable-crash-reporter': false,
  'crash-reporter-id': '84c318ac-ed06-4de5-9f3b-18a05a4b1f85',
  'skip-add-to-recently-opened': false,
  'unity-launch': false,
  'open-url': false,
  'file-write': false,
  'file-chmod': false,
  force: false,
  'do-not-sync': false,
  trace: false,
  'force-user-env': false,
  'force-disable-user-env': false,
  'open-devtools': false,
  'disable-gpu-sandbox': false,
  '__enable-file-policy': false,
  'enable-coi': false,
  'no-proxy-server': false,
  'no-sandbox': false,
  nolazy: false,
  'force-renderer-accessibility': false,
  'ignore-certificate-errors': false,
  'allow-insecure-localhost': false,
  'disable-dev-shm-usage': false,
  'profile-temp': false,
  logsPath: '/home/rongyang/.config/Code/logs/20240124T174223'
}
[90m[main 2024-01-24T04:42:23.719Z][0m Resolving machine identifier...
[90m[main 2024-01-24T04:42:23.719Z][0m Resolved machine identifier: 7b87c88f8777b3c173d6c34ca181471c1d13d73cdbd53f641ad499b6790ecca4
[90m[main 2024-01-24T04:42:23.719Z][0m Main->SharedProcess#connect
[90m[main 2024-01-24T04:42:23.721Z][0m [File Watcher (node.js)] Started watching: '/home/rongyang/.config/Code/User'
[90m[main 2024-01-24T04:42:23.722Z][0m [File Watcher (node.js)] Started watching: '/home/rongyang/.config/Code/User/settings.json'
[90m[main 2024-01-24T04:42:23.722Z][0m StorageMainService: creating application storage
[90m[main 2024-01-24T04:42:23.725Z][0m ElectronURLListener: waiting for window to be ready to handle URLs...
[90m[main 2024-01-24T04:42:23.725Z][0m lifecycle (main): phase changed (value: 2)
[90m[main 2024-01-24T04:42:23.725Z][0m windowsManager#open
[90m[main 2024-01-24T04:42:23.726Z][0m windowsManager#open pathsToOpen [
  {
    workspace: { id: '3e23fe1571390f3cae389d59d977dc32', uri: [$] },
    type: 2,
    exists: true
  }
]
[90m[main 2024-01-24T04:42:23.726Z][0m windowsManager#doOpenFolderOrWorkspace {
  folderOrWorkspace: {
    workspace: { id: '3e23fe1571390f3cae389d59d977dc32', uri: [$] },
    type: 2,
    exists: true
  },
  filesToOpen: undefined
}
[90m[main 2024-01-24T04:42:23.726Z][0m IPC Object URL: Registered new channel vscode:d7e87fda-5f89-4cb5-9926-57e1192133ff.
[90m[main 2024-01-24T04:42:23.727Z][0m window#validateWindowState: validating window state on 3 display(s) { mode: 1, x: 76, y: 178, width: 1717, height: 902 }
[90m[main 2024-01-24T04:42:23.727Z][0m window#validateWindowState: multi-monitor working area { x: 0, y: 0, width: 1920, height: 1080 }
[90m[main 2024-01-24T04:42:23.727Z][0m window#ctor: using window state { mode: 1, x: 76, y: 178, width: 1717, height: 902 }
[90m[main 2024-01-24T04:42:23.760Z][0m window#load: attempt to load window (id: 1)
[90m[main 2024-01-24T04:42:23.764Z][0m StorageMainService: creating workspace storage (3e23fe1571390f3cae389d59d977dc32)
[90m[main 2024-01-24T04:42:23.764Z][0m windowsManager#open used window count 1 (workspacesToOpen: 0, foldersToOpen: 1, emptyToRestore: 0, emptyToOpen: 0)
[90m[main 2024-01-24T04:42:23.764Z][0m lifecycle (main): phase changed (value: 3)
[90m[main 2024-01-24T04:42:23.764Z][0m resolveShellEnv(): skipped (VSCODE_CLI is set)
[90m[main 2024-01-24T04:42:23.765Z][0m update#setState idle
[90m[main 2024-01-24T04:42:26.267Z][0m lifecycle (main): phase changed (value: 4)
[90m[main 2024-01-24T04:42:53.765Z][0m update#checkForUpdates, state =  idle
[90m[main 2024-01-24T04:42:53.765Z][0m update#setState checking for updates
[90m[main 2024-01-24T04:42:53.765Z][0m resolveShellEnv(): skipped (VSCODE_CLI is set)
[91m[main 2024-01-24T04:42:53.843Z][0m CodeWindow: detected unresponsive
[90m[main 2024-01-24T04:42:58.119Z][0m update#setState idle
```

**Comment by tejas-2232** [comment]
> On Ubuntu 22.04 with `v1.82`, I experienced the same issue as @stodge with the same log entry regarding GitHub authentication. When started with `code --password-store="basic"`, as suggested, VSCode successfully starts up.

what this command does?

**Comment by StephenCarlson** [comment]
This is still happening in 2024. I had to make the entry in the ~/.vscode/argv.json file as stated above in https://github.com/microsoft/vscode/issues/188397#issuecomment-1722174241 . If I had not found this issue, VSCode would still be freezing for me with no idea how to fix it. This is on a fresh Ubuntu 22.04 install, the only unique difference is that I have automatic login, no password entry on boot, hence the keyring system is atypical to the normal environment that the developers and most tests are testing in.

**Comment by acce245** [comment]
I was looking through my verbose run of code.
`code --verbose`
and noticed 
`
libGL error: MESA-LOADER: failed to open iris: /usr/lib/dri/iris_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
[7629:0204/084601.146044:ERROR:angle_platform_impl.cc(44)] Display.cpp:1056 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
`

I fixed this portion by simply 
`sudo apt-get install libgl1-mesa-glx`
which I didn't have for some reason.

This doesn't fully resolve the issue. 

I did a bit more sniffing around, and after 
`sudo apt-get install mesa-utils'
it resolved the issues and now vscode just works.

I did coincidentally 
`sudo add-apt-repository ppa:ubuntu-x-swat/updates` 
in the same step so this repo might be useful to some people but I don't think it caused the fix with mesa iris and swrast.  

I'm not sure if this fix will help anyone but I suspect mesa-utils was the primary fix for some reason. 

Here's the [full verbose of the first run](https://pastebin.com/ER6mwJ5C) with the problem. That is a pastebin link.  github doesn't like formatting that long text or I would dump it here.

The verbose after is now:

`[25382:0204/093304.542566:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.
[main 2024-02-04T14:33:04.607Z] [File Watcher (node.js)] Request to start watching: /home/acce245/.config/Code/User (excludes: <none>, includes: <all>, correlationId: <none>),/home/acce245/.config/Code/User/settings.json (excludes: <none>, includes: <all>, correlationId: <none>)
[main 2024-02-04T14:33:04.610Z] Sending env to running instance...
[main 2024-02-04T14:33:04.611Z] [File Watcher (node.js)] Started watching: '/home/acce245/.config/Code/User'
[main 2024-02-04T14:33:04.611Z] [File Watcher (node.js)] Started watching: '/home/acce245/.config/Code/User/settings.json'
[main 2024-02-04T14:33:04.642Z] Sent env to running instance. Terminating...
[main 2024-02-04T14:33:04.642Z] Lifecycle#kill()
[main 2024-02-04T14:33:04.642Z] Lifecycle#onWillShutdown.fire()
`
and vscode opens instantly.

Again, not sure if this issue is unique to my machine or instance of vscode.
This machine runs Ubuntu 22.04.3 and I have the same installed on another laptop.  After running the same commands and doing apt and apt-get update, upgrade and dist-upgrade, they both seem to open as quick as expected.  The other laptop still shows some errors in the verbose related to mesa but it still loads in a fraction of a second rather than hanging.

It looks like this mesa erroring might be a larger issue across multiple programs and various distros.  Again, I'm not sure what fixed the issue, but I'm almost positive it was some combination of these steps. Hopefully this will point someone smarter than me in the direction of a proper fix.

**Comment by EduartePaiva** [comment]
I had the same issue on Ubuntu 22.04.3 LTS, doing the "password-store" : "basic" solved the freezing but I don't know what this setting does

**Comment by acce245** [comment]
I updated my drivers using Nvidia drivers rather than the open source options and this seemed to resolve the issue finally.  I'm guessing it has something to do with the specific libraries it's looking for not appearing as expected.  If you have a graphics card and use the proprietary driver rather than a FOSS alternative, that may solve the issue.  If you cannot use it this may not work, but I have not reason I can't use it, I just didn't realize it hadn't loaded on either of those instances I used.  Perhaps try that if possible?

**Comment by SerenaAugtera** [comment]
I have the same "not responding on launching" issue for weeks, and I finally decided to do something with it. I tried uninstall and reinstall vscode, also tried the `code password-store="basic"`. None of these works...

**Comment by mdenty** [comment]
Same issue here on Noble (early 24.04).

**Comment by perlun** [comment]
Same problem for me on Debian Bookworm. Adding the `--password-store="basic"` parameter works (either directly or via `~/.vscode/argv.json`).

I have a laptop with dual Intel and nVidia graphics. Currently using the built-in Intel graphics and preferring to not enable the nVidia card (& drivers) since I've had issues with suspend/resume on my previous (similar) machine.

```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) Graphics (RPL-P) (0xa7a0)
    Version: 22.3.6
    Accelerated: yes
    Video memory: 31764MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) Graphics (RPL-P)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 22.3.6
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.3.6
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 22.3.6
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

**Comment by jyap808** [comment]
Replicated this issue on Arch Linux. Last verbose line:
```
[48891:0402/133510.060815:INFO:CONSOLE(651)] "%cTRACE color: #888 [mainThreadSecretState] Getting password for vscode.github-authentication extension:  github.auth", source: vscode-file://vscode-app/usr/lib/code/out/vs/workbench/workbench.desktop.main.js (651)
```

code --version
```
1.87.2
863d2581ecda6849923a2118d93a088b0745d9d6
x64
```

uname -r
```
6.8.2-arch2-1
```

Workarounds helped here.

**Comment by perlun** [comment]
I think my problem may have been this: copying my home directory from one machine to another. The user name was the same, but the login password was different (I think). This caused some kind of inconsistency with the `gnome-keyring` or whatever. I got it fixed somehow, and this may have been the real "root cause" of the problem I was experiencing a few weeks ago with this after moving to a new laptop.

**Comment by All3xJ** [comment]
I still have this issue in Ubuntu 24.04.` --password-store="basic"` fixes

**Comment by AdnanHodzic** [comment]
I also have this problem on Ubuntu 24.04, how I got it working is [by referencing to a 4 year old post on snapcraft](https://forum.snapcraft.io/t/vscode-does-not-open/16339/13?u=ahodzic), which is to reboot. Inconvenient and annoying to say the least.

**Comment by Tomfez** [comment]
I have the same behavior on a fresh Linux Mint 22 install. Adding  `--password-store="basic"` fixes the issue but I still don't know why it is the case.

**Comment by minaelee** [comment]
Same issue. Resolved with `--password-store="basic"` like everyone else. Thanks!

Fresh Ubuntu 24.04 LTS installation on an HP laptop
VS Code version: 1.92.1 - installed with amd64.deb using apt

```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) HD Graphics 5500 (BDW GT2) (0x1616)
    Version: 24.0.9
    Accelerated: yes
    Video memory: 7815MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) HD Graphics 5500 (BDW GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 24.0.9-0ubuntu0.1
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 24.0.9-0ubuntu0.1
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 24.0.9-0ubuntu0.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

**Comment by Scavanger** [comment]
I had the same problem under Kubuntu 24.04, all open suggestions did not help.
I then installed the nvidia drivers via `sudo ubuntu-drivers list`, now VScode works fine

**Comment by xianghongai** [comment]
My problem happened after Ubuntu reset the user operating system password~

`code --verbose`

```
[25485:1020/125018.685190:ERROR:nss_util.cc(357)] After loading Root Certs, loaded==false: NSS error code: -8018
[main 2024-10-20T04:50:18.753Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[main 2024-10-20T04:50:18.755Z] [network] #1: https://update.code.visualstudio.com/api/update/linux-x64/stable/384ff7382de624fb94dbaf6da11977bba1ecd427 - begin undefined X7 { b: {}, a: [Object: null prototype] {} }
[25485:1020/125019.030373:INFO:CONSOLE(25)] "Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!
    at Hni.add (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:25:1553)
    at vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:2066:25399", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (25)
[25485:1020/125019.057267:INFO:CONSOLE(25)] "Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!
    at Hni.add (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:25:1553)
    at vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:2066:25399", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (25)
[25485:1020/125019.076413:INFO:CONSOLE(35)] "%cDEBUG background: #eee; color: #888 [Registering Log Channel] File does not exist. Waiting for 1s to retry. file:///home/xianghongai/.config/Code/logs/20241020T125017/window1/notebook.rendering.log", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (35)
[25485:1020/125019.080665:INFO:CONSOLE(35)] "%cTRACE color: #888 [mainThreadSecretState] Getting password for vscode.microsoft-authentication extension:  microsoft.login.keylist", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (35)
[main 2024-10-20T04:50:19.080Z] [EncryptionMainService] Checking if encryption is available...
```

**Comment by IS2511** [comment]
I didn't want to change my password store for VSCode since I know that my keyring worked in the past and should work now. So I figured the GNOME keyring daemon is probably in some weird state. I checked if it was running - `pgrep gnome-keyring`, sure enough it is, and the PID is pretty low - 2428, must be running from boot. I do remember being asked to unlock my keyring on login. Let's just kill the daemon and restart it. Close VSCode > `kill "$(pgrep gnome-keyring)"` > Open VSCode again. After starting VSCode I was once again asked to unlock my keyring (presumably VSCode triggered the keyring daemon to start again), so I did. Finally, VSCode didn't hang.

TLDR: Close VSCode > `kill "$(pgrep gnome-keyring)"` > Open VSCode. (this will restart the `gnome-keyring-daemon`)

No idea why the whole VSCode window hangs when it (presumably) can't get an answer from the GNOME keyring daemon. Doesn't seem like intended behavior. Funnily enough, all the other VSCode stuff seems to run fine, I can see rust-analyzer building the crate and all other extensions doing what they are supposed to do, yet the window stops responding a couple seconds after VSCode starts.

<details><summary>Boring environment details (spoiler)</summary>

```
$ code --version
1.95.2
e8653663e8840adaf45af01eab5c627a5af81807
x64
$ nixos-version 
24.11.711150.314e12ba369c (Vicuna)
$ grep vscode /etc/nixos/configuration.nix 
    vscode-fhs
$ date --utc --iso-8601 
2024-12-18
```

</details>
